merge dev to master

SaPer663 opened the pull request.

SaPer663 committed:
fix: feat/csv_import перносит импортируемую бибилиотеку в секцию сторонних приложений

SaPer663 committed:
Merge pull request #41 from online-furniture-store/feat/csv_import

CSV import

SaPer663 committed:
Merge pull request #44 from online-furniture-store/feat/popular

Добавлены топ шестерка товаров

Markello93 committed:
models_discount

Markello93 committed:
models_discount

SaPer663 committed:
fix: feat/product разрешает конфликт слияния с веткой dev

SaPer663 committed:
fix: feat/product удаляет неиспользуемую библиотеку из зависимостей

SaPer663 committed:
Merge pull request #45 from online-furniture-store/feat/product

models_discount

Commit:
Исправление замечаний

Commit:
Правки замечаний

Commit:
Настроены способы доставки, список доставок, склад

dependabot[bot] committed:
build(deps): bump traefik in /compose/production/traefik

Bumps traefik from 2.10.1 to 2.10.3.

---
updated-dependencies:
- dependency-name: traefik
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump mypy from 1.3.0 to 1.4.0

Bumps [mypy](https://github.com/python/mypy) from 1.3.0 to 1.4.0.
- [Commits](https://github.com/python/mypy/compare/v1.3.0...v1.4.0)

---
updated-dependencies:
- dependency-name: mypy
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

SaPer663 committed:
Merge pull request #47 from online-furniture-store/dependabot/pip/mypy-1.4.0

build(deps): bump mypy from 1.3.0 to 1.4.0

SaPer663 committed:
Merge pull request #46 from online-furniture-store/dependabot/docker/compose/production/traefik/traefik-2.10.3

build(deps): bump traefik from 2.10.1 to 2.10.3 in /compose/production/traefik

Artek22 committed:
admin image preview

Artek22 committed:
admin image preview

SaPer663 committed:
fix: feat/admin_image_preview переносит функцию превью картинки пеосле определения атрибутов

SaPer663 committed:
fix: feat/admin_image_preview исправляет конфликт слияния

SaPer663 committed:
Merge pull request #48 from online-furniture-store/feat/admin_image_preview

admin image preview

mikepavlos committed:
Создание файлов базы

Commit:
Исправление моделей

Commit:
Merge branch 'dev' into feat/orders

Commit:
Мерж dev

mikepavlos committed:
Создание файлов базы

SaPer663 committed:
Merge pull request #49 from online-furniture-store/feat/database

Создание файлов базы

mikepavlos committed:
рефакт db_products.json

SaPer663 committed:
Merge pull request #50 from online-furniture-store/feat/database

рефакт db_products.json

SaPer663 committed:
feat: feat/product-api добавляет вывод подробной информации о продукте и его стоимости

mikepavlos committed:
Merge pull request #51 from online-furniture-store/feat/product-api

feat: feat/product-api добавляет вывод подробной информации о продукт…

Markello93 committed:
related_name+strings

Commit:
Настроена все модели, кроме поля total_cost в Order

mikepavlos committed:
API для Collection

SaPer663 committed:
feat: feat/extend-serializer-fields-for-popular-and-favorite-products добавляет поля в не полный сериализатор и испарвляет ошибку подсчёта цены со скидкой

SaPer663 committed:
feat: feat/extend-serializer-fields-for-popular-and-favorite-products добавляет поля в не полный сериализатор и испарвляет ошибку подсчёта цены со скидкой

SaPer663 committed:
Merge pull request #52 from online-furniture-store/feat/product

related_name+strings

SaPer663 committed:
Merge branch 'feat/extend-serializer-fields-for-popular-and-favorite-products' of github.com:online-furniture-store/online_furniture_store_backend into feat/extend-serializer-fields-for-popular-and-favorite-products

SaPer663 committed:
Merge pull request #54 from online-furniture-store/feat/extend-serializer-fields-for-popular-and-favorite-products

feat: feat/extend-serializer-fields-for-popular-and-favorite-products…

SaPer663 committed:
feat: feat/setup-cors добавляет разрешение на обработку соединений со всех адресов

mikepavlos committed:
Merge pull request #55 from online-furniture-store/feat/setup-cors

feat: feat/setup-cors добавляет разрешение на обработку соединений со…

mikepavlos committed:
Merge branch 'dev' into feat/collections

# Conflicts:
#	apps/product/serializers.py

mikepavlos committed:
API для Collection

mikepavlos committed:
API для Collection, исправлена очепятка

SaPer663 committed:
Merge pull request #53 from online-furniture-store/feat/collections

API для Collection

Commit:
Правки замечаний

Commit:
merge dev

Commit:
merge dev

Commit:
Добавлено отображение всех заказов пользователя

dependabot[bot] committed:
build(deps): bump drf-spectacular from 0.26.2 to 0.26.3

Bumps [drf-spectacular](https://github.com/tfranzel/drf-spectacular) from 0.26.2 to 0.26.3.
- [Release notes](https://github.com/tfranzel/drf-spectacular/releases)
- [Changelog](https://github.com/tfranzel/drf-spectacular/blob/master/CHANGELOG.rst)
- [Commits](https://github.com/tfranzel/drf-spectacular/compare/0.26.2...0.26.3)

---
updated-dependencies:
- dependency-name: drf-spectacular
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

SaPer663 committed:
fix: debug/product-image добавляет директорию для статики и медиа

SaPer663 committed:
Merge pull request #56 from online-furniture-store/dependabot/pip/drf-spectacular-0.26.3

build(deps): bump drf-spectacular from 0.26.2 to 0.26.3

SaPer663 committed:
Merge pull request #57 from online-furniture-store/debug/product-image

fix: debug/product-image добавляет директорию для статики и медиа

SaPer663 committed:
fix: debug/product-image изменяет путь хранения медиа файлов на сервере и добавляет переменную окружения с этим путём

SaPer663 committed:
Merge pull request #58 from online-furniture-store/debug/product-image

fix: debug/product-image изменяет путь хранения медиа файлов на серве…

mikepavlos committed:
Редактирование связанных таблиц в админке; Автозаполнение slug

mikepavlos committed:
Merge branch 'dev' into feat/products_admin

mikepavlos committed:
Исправление метода представления __str__ корзины на email

Commit:
Правки

Commit:
Правки

Commit:
Настроено получение общей строимости заказа

SaPer663 committed:
Merge pull request #59 from online-furniture-store/feat/products_admin

Feat/products admin

dependabot[bot] committed:
build(deps): bump mypy from 1.4.0 to 1.4.1

Bumps [mypy](https://github.com/python/mypy) from 1.4.0 to 1.4.1.
- [Commits](https://github.com/python/mypy/compare/v1.4.0...v1.4.1)

---
updated-dependencies:
- dependency-name: mypy
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump pytest from 7.3.2 to 7.4.0

Bumps [pytest](https://github.com/pytest-dev/pytest) from 7.3.2 to 7.4.0.
- [Release notes](https://github.com/pytest-dev/pytest/releases)
- [Changelog](https://github.com/pytest-dev/pytest/blob/main/CHANGELOG.rst)
- [Commits](https://github.com/pytest-dev/pytest/compare/7.3.2...7.4.0)

---
updated-dependencies:
- dependency-name: pytest
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump redis from 4.5.5 to 4.6.0

Bumps [redis](https://github.com/redis/redis-py) from 4.5.5 to 4.6.0.
- [Release notes](https://github.com/redis/redis-py/releases)
- [Changelog](https://github.com/redis/redis-py/blob/master/CHANGES)
- [Commits](https://github.com/redis/redis-py/compare/v4.5.5...v4.6.0)

---
updated-dependencies:
- dependency-name: redis
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Artek22 committed:
Cleaning garbage

SaPer663 committed:
Merge pull request #63 from online-furniture-store/dependabot/pip/redis-4.6.0

build(deps): bump redis from 4.5.5 to 4.6.0

SaPer663 committed:
Merge pull request #62 from online-furniture-store/dependabot/pip/pytest-7.4.0

build(deps): bump pytest from 7.3.2 to 7.4.0

SaPer663 committed:
Merge pull request #61 from online-furniture-store/dependabot/pip/mypy-1.4.1

build(deps): bump mypy from 1.4.0 to 1.4.1

Markello93 committed:
cart

Artek22 committed:
Views, api_router UserViewSet delete

Commit:
Исправлена ошибка в изменении заказа

Commit:
Исправлена ошибка в изменении заказа

Commit:
Исправлена ошибка в изменении заказа

dependabot[bot] committed:
build(deps): bump django-stubs from 4.2.1 to 4.2.2

Bumps [django-stubs](https://github.com/typeddjango/django-stubs) from 4.2.1 to 4.2.2.
- [Release notes](https://github.com/typeddjango/django-stubs/releases)
- [Commits](https://github.com/typeddjango/django-stubs/compare/4.2.1...4.2.2)

---
updated-dependencies:
- dependency-name: django-stubs
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump djangorestframework-stubs from 3.14.1 to 3.14.2

Bumps [djangorestframework-stubs](https://github.com/typeddjango/djangorestframework-stubs) from 3.14.1 to 3.14.2.
- [Release notes](https://github.com/typeddjango/djangorestframework-stubs/releases)
- [Commits](https://github.com/typeddjango/djangorestframework-stubs/compare/3.14.1...3.14.2)

---
updated-dependencies:
- dependency-name: djangorestframework-stubs
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

SaPer663 committed:
Merge pull request #65 from online-furniture-store/fix/users

Cleaning garbage

SaPer663 committed:
Merge pull request #68 from online-furniture-store/dependabot/pip/djangorestframework-stubs-3.14.2

build(deps): bump djangorestframework-stubs from 3.14.1 to 3.14.2

SaPer663 committed:
Merge pull request #67 from online-furniture-store/dependabot/pip/django-stubs-4.2.2

build(deps): bump django-stubs from 4.2.1 to 4.2.2

SaPer663 committed:
fix: разрешает конфликт слияния

SaPer663 committed:
fix: feat/cart-improvements исправляет инициализацию корзины для анонимного пользователя

SaPer663 committed:
Merge pull request #69 from online-furniture-store/feat/cart-improvements

Изменяет механику работы корзины

SaPer663 committed:
feat:  feat/add-available-quantity-and-base64-image-field изменяет тип поля изображения товара, добавляет поле количества доступного товара и контекст в сериализаторы товаров

SaPer663 committed:
Merge pull request #70 from online-furniture-store/feat/add-available-quantity-and-base64-image-field

feat:  feat/add-available-quantity-and-base64-image-field изменяет ти…

SaPer663 committed:
fix: fix/user-cart-view-function исправляет функционал получения данных о карзине пользвателя

mikepavlos committed:
Доработана функциональность запросов в апи; Актуализировано популярное

mikepavlos committed:
Merge remote-tracking branch 'origin/feat/orders' into feat/orders

# Conflicts:
#	apps/orders/serializers.py

Markello93 committed:
Merge pull request #71 from online-furniture-store/fix/user-cart-view-function

fix: fix/user-cart-view-function исправляет функционал получения данн…

mikepavlos committed:
merge

SaPer663 committed:
Merge pull request #40 from online-furniture-store/feat/orders

Feat/orders

SaPer663 committed:
fix: fix/available_quantity-field добавляет потерянный, при слиянии код

SaPer663 committed:
Merge pull request #72 from online-furniture-store/fix/available_quantity-field

fix: fix/available_quantity-field добавляет потерянный, при слиянии код

SaPer663 committed:
fix fix/fix-del-item-error исправляет ошибку в url

mikepavlos committed:
база с заказами и складом, избранное.

mikepavlos committed:
Merge pull request #74 from online-furniture-store/feat/database_oreders

база с заказами и складом, избранное.

SaPer663 committed:
Merge pull request #73 from online-furniture-store/fix/fix-del-item-error

fix fix/fix-del-item-error исправляет ошибку в url

KoshelevAleksandr committed:
Изменен родительский класс для моделе Admin

SaPer663 committed:
Merge pull request #75 from online-furniture-store/feat/orders

Изменен родительский класс для классов Admin

dependabot[bot] committed:
build(deps): bump django-stubs from 4.2.2 to 4.2.3

Bumps [django-stubs](https://github.com/typeddjango/django-stubs) from 4.2.2 to 4.2.3.
- [Release notes](https://github.com/typeddjango/django-stubs/releases)
- [Commits](https://github.com/typeddjango/django-stubs/compare/4.2.2...4.2.3)

---
updated-dependencies:
- dependency-name: django-stubs
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

mikepavlos committed:
Корректировка логики популярного; CORS_credentials; Обновление data/csv.

Markello93 committed:
Merge pull request #77 from online-furniture-store/fix/popular

Корректировка логики популярного; CORS_credentials; Обновление data/csv.

SaPer663 committed:
Merge pull request #76 from online-furniture-store/dependabot/pip/django-stubs-4.2.3

build(deps): bump django-stubs from 4.2.2 to 4.2.3

SaPer663 committed:
fix: fix/fix-popular-products-image-paths добавляет контекс в сериализатор популярных товаров

SaPer663 committed:
Merge pull request #78 from online-furniture-store/fix/fix-popular-products-image-paths

fix: fix/fix-popular-products-image-paths добавляет контекс в сериали…

Markello93 committed:
router

Markello93 committed:
serializers+views_dev

SaPer663 committed:
feat: feat/expand-cart-item-info добавляет поля в данные о товарах в корзине

SaPer663 committed:
Merge pull request #79 from online-furniture-store/feat/expand-cart-item-info

feat: feat/expand-cart-item-info добавляет поля в данные о товарах в …

Markello93 committed:
Merge branch 'dev' into feat/product

mikepavlos committed:
Доработка методов заказов create, update для корректного обновления склада, расчета total_cost.

mikepavlos committed:
Доработка представления заказов. Только создание, получение.

SaPer663 committed:
feat: feat/expand-cart-item-info добавляет в ответ вывод информации о товарах в корзине при удалении товара из корзины

mikepavlos committed:
Merge pull request #80 from online-furniture-store/feat/expand-cart-item-info

feat: feat/expand-cart-item-info добавляет в ответ вывод информации о…

mikepavlos committed:
Merge branch 'dev' into fix/order_updates_storehouse

mikepavlos committed:
Рефакторинг сериализаторов заказов. Доработка представления заказов. Только создание, получение.

dependabot[bot] committed:
build(deps): bump pillow from 9.5.0 to 10.0.0

Bumps [pillow](https://github.com/python-pillow/Pillow) from 9.5.0 to 10.0.0.
- [Release notes](https://github.com/python-pillow/Pillow/releases)
- [Changelog](https://github.com/python-pillow/Pillow/blob/main/CHANGES.rst)
- [Commits](https://github.com/python-pillow/Pillow/compare/9.5.0...10.0.0)

---
updated-dependencies:
- dependency-name: pillow
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

SaPer663 committed:
fix: fix/order_updates_storehouse исправляет ошибку в сериализаторе

SaPer663 committed:
Merge pull request #81 from online-furniture-store/fix/order_updates_storehouse

Fix/order updates storehouse

SaPer663 committed:
Merge pull request #82 from online-furniture-store/dependabot/pip/pillow-10.0.0

build(deps): bump pillow from 9.5.0 to 10.0.0

SaPer663 committed:
fix: feat/expand-cart-item-info изменяет статус код ответа при удалении товара из корзины

SaPer663 committed:
fix: feat/expand-cart-item-info изменяет статус код ответа при удалении товара из корзины

SaPer663 committed:
Merge branch 'feat/expand-cart-item-info' of github.com:online-furniture-store/online_furniture_store_backend into feat/expand-cart-item-info

Markello93 committed:
related

SaPer663 committed:
Merge pull request #83 from online-furniture-store/feat/expand-cart-item-info

fix: feat/expand-cart-item-info изменяет статус код ответа при удален…

Markello93 committed:
serializers

Markello93 committed:
migrations

Markello93 committed:
delete perm

Markello93 committed:
verbose_names

Markello93 committed:
migrations_verbose

SaPer663 committed:
Merge pull request #85 from online-furniture-store/feat/product

Feat/product

dependabot[bot] committed:
build(deps): bump django from 4.2.2 to 4.2.3

Bumps [django](https://github.com/django/django) from 4.2.2 to 4.2.3.
- [Commits](https://github.com/django/django/compare/4.2.2...4.2.3)

---
updated-dependencies:
- dependency-name: django
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Markello93 committed:
filters

SaPer663 committed:
Merge pull request #86 from online-furniture-store/dependabot/pip/django-4.2.3

build(deps): bump django from 4.2.2 to 4.2.3

Markello93 committed:
requirements_filter_comment

SaPer663 committed:
build: feat/setup_ci_cd изменяет конфиг ci/cd

SaPer663 committed:
build: feat/setup_ci_cd изменяет конфиг ci/cd

SaPer663 committed:
Merge branch 'feat/setup_ci_cd' of github.com:online-furniture-store/online_furniture_store_backend into feat/setup_ci_cd

SaPer663 committed:
build: feat/setup_ci_cd добавляет деплой при пуше в эту ветку

SaPer663 committed:
build: feat/setup_ci_cd именяет конфиг github flow на развёртывание в ветке dev

SaPer663 committed:
build: feat/setup_ci_cd исправляет опечатку в названии ветки

mikepavlos committed:
Merge pull request #88 from online-furniture-store/feat/setup_ci_cd

Feat/setup ci cd

mikepavlos committed:
Создает пользователя и доставку при оформлении заказа.

Markello93 committed:
user_serializer+urls

mikepavlos committed:
Логика создания пользователя перенесена в сериализатор OrderWriteSerializer.

mikepavlos committed:
Обработка на дублирование товаров в одном заказе.

mikepavlos committed:
Merge pull request #89 from online-furniture-store/feat/order_anonymous

Создает пользователя и доставку при оформлении заказа.

mikepavlos committed:
Устраняет ошибку создания заказа с авторизованным пользователем.

SaPer663 committed:
build: feat/setup_ci_cd изменяет конфиг развертывания прокси сервера и его конфиг

SaPer663 committed:
build: feat/setup_ci_cd изменяет конфиг развертывания прокси сервера и его конфиг

mikepavlos committed:
Merge pull request #91 from online-furniture-store/fix/order_anonymous

Устраняет ошибку создания заказа с авторизованным пользователем.

SaPer663 committed:
Merge branch 'feat/setup_ci_cd' of github.com:online-furniture-store/online_furniture_store_backend into feat/setup_ci_cd

SaPer663 committed:
Merge pull request #92 from online-furniture-store/feat/setup_ci_cd

Feat/setup ci cd

dependabot[bot] committed:
build(deps): bump django-coverage-plugin from 3.0.0 to 3.1.0

Bumps [django-coverage-plugin](https://github.com/nedbat/django_coverage_plugin) from 3.0.0 to 3.1.0.
- [Release notes](https://github.com/nedbat/django_coverage_plugin/releases)
- [Commits](https://github.com/nedbat/django_coverage_plugin/compare/v3.0.0...v3.1.0)

---
updated-dependencies:
- dependency-name: django-coverage-plugin
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump django-cors-headers from 4.1.0 to 4.2.0

Bumps [django-cors-headers](https://github.com/adamchainz/django-cors-headers) from 4.1.0 to 4.2.0.
- [Changelog](https://github.com/adamchainz/django-cors-headers/blob/main/CHANGELOG.rst)
- [Commits](https://github.com/adamchainz/django-cors-headers/compare/4.1.0...4.2.0)

---
updated-dependencies:
- dependency-name: django-cors-headers
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump black from 23.3.0 to 23.7.0

Bumps [black](https://github.com/psf/black) from 23.3.0 to 23.7.0.
- [Release notes](https://github.com/psf/black/releases)
- [Changelog](https://github.com/psf/black/blob/main/CHANGES.md)
- [Commits](https://github.com/psf/black/compare/23.3.0...23.7.0)

---
updated-dependencies:
- dependency-name: black
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

SaPer663 committed:
Merge pull request #95 from online-furniture-store/dependabot/pip/black-23.7.0

build(deps): bump black from 23.3.0 to 23.7.0

SaPer663 committed:
Merge pull request #94 from online-furniture-store/dependabot/pip/django-cors-headers-4.2.0

build(deps): bump django-cors-headers from 4.1.0 to 4.2.0

SaPer663 committed:
Merge pull request #93 from online-furniture-store/dependabot/pip/django-coverage-plugin-3.1.0

build(deps): bump django-coverage-plugin from 3.0.0 to 3.1.0

Markello93 committed:
change_serilizer_view

Markello93 committed:
Merge pull request #87 from online-furniture-store/feat/product

filters

Markello93 committed:
fix_users

SaPer663 committed:
build: feat/deploy-frontend-backend-together добавляет конфиг workflow для развёртывания front и back частей проекта

SaPer663 committed:
build: feat/deploy-frontend-backend-together добавляет конфиг workflow для развёртывания front и back частей проекта

SaPer663 committed:
build: feat/deploy-frontend-backend-together исправляет опечатку в имени ветки

SaPer663 committed:
build: feat/deploy-frontend-backend-together исправляет опечатку в имени ветки

Markello93 committed:
fix docstring

dependabot[bot] committed:
build(deps): bump django-filter from 22.1 to 23.2

Bumps [django-filter](https://github.com/carltongibson/django-filter) from 22.1 to 23.2.
- [Release notes](https://github.com/carltongibson/django-filter/releases)
- [Changelog](https://github.com/carltongibson/django-filter/blob/main/CHANGES.rst)
- [Commits](https://github.com/carltongibson/django-filter/compare/22.1...23.2)

---
updated-dependencies:
- dependency-name: django-filter
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

SaPer663 committed:
test repository_dispatch

SaPer663 committed:
Merge pull request #98 from SaPer663/feat/deploy-frontend-backend-together

test repository_dispatch

SaPer663 committed:
Revert "test repository_dispatch"

SaPer663 committed:
Merge pull request #99 from online-furniture-store/revert-98-feat/deploy-frontend-backend-together

Revert "test repository_dispatch"

SaPer663 committed:
Merge pull request #97 from online-furniture-store/dependabot/pip/django-filter-23.2

build(deps): bump django-filter from 22.1 to 23.2

mikepavlos committed:
Переносит генерацию пароля в UserManager.

SaPer663 committed:
fix: feat/deploy-frontend-backend-together удаляет не используемый подмодуль

Markello93 committed:
fix users

mikepavlos committed:
Merge branch 'fix/users' into fix/gen_parol

Markello93 committed:
Merge pull request #96 from online-furniture-store/fix/users

fix_users

SaPer663 committed:
build: feat/deploy-frontend-backend-together настраивает совместный деплой фронта и бека

SaPer663 committed:
fix: feat/deploy-frontend-backend-together разрешает конфликт слияния

mikepavlos committed:
Merge remote-tracking branch 'origin/dev' into fix/gen_parol

SaPer663 committed:
fix: feat/deploy-frontend-backend-together добавляет отступ внутри блока редиректа с http -> https

SaPer663 committed:
Merge pull request #101 from online-furniture-store/feat/deploy-frontend-backend-together

Feat/deploy frontend backend together

SaPer663 committed:
fix: feat/deploy-frontend-backend-together исправляет изменения ошибочно попавшие в коммит при разрешении конфликта слияния

mikepavlos committed:
Добавляет в UserSerializer работу с паролем при создании пользователя.

SaPer663 committed:
Merge pull request #102 from online-furniture-store/feat/deploy-frontend-backend-together

fix: feat/deploy-frontend-backend-together исправляет изменения ошибо…

mikepavlos committed:
Ссылка на сайт и email сайта в константы настроек и переменные окружения.

mikepavlos committed:
Merge branch 'dev' into fix/gen_parol

mikepavlos committed:
Merge branch 'dev' into fix/gen_parol

SaPer663 committed:
Merge pull request #100 from online-furniture-store/fix/gen_parol

Переносит генерацию пароля в UserManager.

SaPer663 committed:
fix: feat/deploy-frontend-backend-together  исправляет не правильные инструкции в конфиге workflow

SaPer663 committed:
Merge pull request #103 from online-furniture-store/feat/deploy-frontend-backend-together

fix: feat/deploy-frontend-backend-together  исправляет не правильные …

Markello93 committed:
images_admin_type

Commit:
Исправлено получение цены товара в заказе

Markello93 committed:
images_admin_serializers

dependabot[bot] committed:
build(deps): bump uvicorn[standard] from 0.22.0 to 0.23.0

Bumps [uvicorn[standard]](https://github.com/encode/uvicorn) from 0.22.0 to 0.23.0.
- [Release notes](https://github.com/encode/uvicorn/releases)
- [Changelog](https://github.com/encode/uvicorn/blob/master/CHANGELOG.md)
- [Commits](https://github.com/encode/uvicorn/compare/0.22.0...0.23.0)

---
updated-dependencies:
- dependency-name: uvicorn[standard]
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Commit:
Исправлен расчет стоимости товаров в заказе

dependabot[bot] committed:
build(deps): bump gunicorn from 20.1.0 to 21.0.1

Bumps [gunicorn](https://github.com/benoitc/gunicorn) from 20.1.0 to 21.0.1.
- [Release notes](https://github.com/benoitc/gunicorn/releases)
- [Commits](https://github.com/benoitc/gunicorn/compare/20.1.0...21.0.1)

---
updated-dependencies:
- dependency-name: gunicorn
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

mikepavlos committed:
Изменение модели доставки на поля в запросе.

SaPer663 committed:
feat: feat/add-other-color-products-info добавляет информацию в выводе товара, о таких же товарах другого цвета

SaPer663 committed:
feat: feat/add-other-color-products-info добавляет вывод информации по подобным товарам

SaPer663 committed:
Merge pull request #105 from online-furniture-store/feat/orders

Feat/orders

SaPer663 committed:
Merge pull request #106 from online-furniture-store/dependabot/pip/gunicorn-21.0.1

build(deps): bump gunicorn from 20.1.0 to 21.0.1

SaPer663 committed:
Merge pull request #104 from online-furniture-store/dependabot/pip/uvicorn-standard--0.23.0

build(deps): bump uvicorn[standard] from 0.22.0 to 0.23.0

SaPer663 committed:
Merge pull request #107 from online-furniture-store/fix/delivery

Изменение модели доставки на поля в запросе.

SaPer663 committed:
Merge pull request #111 from online-furniture-store/feat/add-other-color-products-info

Feat/add other color products info

dependabot[bot] committed:
build(deps): bump uvicorn[standard] from 0.23.0 to 0.23.1

Bumps [uvicorn[standard]](https://github.com/encode/uvicorn) from 0.23.0 to 0.23.1.
- [Release notes](https://github.com/encode/uvicorn/releases)
- [Changelog](https://github.com/encode/uvicorn/blob/master/CHANGELOG.md)
- [Commits](https://github.com/encode/uvicorn/compare/0.23.0...0.23.1)

---
updated-dependencies:
- dependency-name: uvicorn[standard]
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump gunicorn from 21.0.1 to 21.1.0

Bumps [gunicorn](https://github.com/benoitc/gunicorn) from 21.0.1 to 21.1.0.
- [Release notes](https://github.com/benoitc/gunicorn/releases)
- [Commits](https://github.com/benoitc/gunicorn/compare/21.0.1...21.1.0)

---
updated-dependencies:
- dependency-name: gunicorn
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

SaPer663 committed:
feat: feat/favorite-product-response добавляет вывод информации по всем избранным товарампользователя  при добавлении и удалении товара в избранное

SaPer663 committed:
Merge pull request #113 from online-furniture-store/dependabot/pip/gunicorn-21.1.0

build(deps): bump gunicorn from 21.0.1 to 21.1.0

SaPer663 committed:
Merge pull request #112 from online-furniture-store/dependabot/pip/uvicorn-standard--0.23.1

build(deps): bump uvicorn[standard] from 0.23.0 to 0.23.1

SaPer663 committed:
Merge pull request #114 from online-furniture-store/feat/favorite-product-response

feat: feat/favorite-product-response добавляет вывод информации по вс…

mikepavlos committed:
Добавляет расчет оставшегося количества товара на складе.

SaPer663 committed:
Merge pull request #115 from online-furniture-store/fix/product_available_quantity

Добавляет расчет оставшегося количества товара на складе.

SaPer663 committed:
fix: fix/email-sending-error исправляет настройки почтового сервера

SaPer663 committed:
Merge pull request #116 from online-furniture-store/fix/email-sending-error

fix: fix/email-sending-error исправляет настройки почтового сервера

dependabot[bot] committed:
build(deps): bump factory-boy from 3.2.1 to 3.3.0

Bumps [factory-boy](https://github.com/FactoryBoy/factory_boy) from 3.2.1 to 3.3.0.
- [Changelog](https://github.com/FactoryBoy/factory_boy/blob/master/docs/changelog.rst)
- [Commits](https://github.com/FactoryBoy/factory_boy/compare/3.2.1...3.3.0)

---
updated-dependencies:
- dependency-name: factory-boy
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump gunicorn from 21.1.0 to 21.2.0

Bumps [gunicorn](https://github.com/benoitc/gunicorn) from 21.1.0 to 21.2.0.
- [Release notes](https://github.com/benoitc/gunicorn/releases)
- [Commits](https://github.com/benoitc/gunicorn/compare/21.1.0...21.2.0)

---
updated-dependencies:
- dependency-name: gunicorn
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

SaPer663 committed:
feat: feat/download-email добавляет ендпойнт для загрузки файла письма с сервера для тестирования функционала

SaPer663 committed:
Merge pull request #119 from online-furniture-store/feat/download-email

feat: feat/download-email добавляет ендпойнт для загрузки файла письм…

SaPer663 committed:
Merge pull request #118 from online-furniture-store/dependabot/pip/gunicorn-21.2.0

build(deps): bump gunicorn from 21.1.0 to 21.2.0

SaPer663 committed:
Merge pull request #117 from online-furniture-store/dependabot/pip/factory-boy-3.3.0

build(deps): bump factory-boy from 3.2.1 to 3.3.0

SaPer663 committed:
fix: fix/popular-products-image-display исправляет отображение изображений товаров в катологе популярных товаров

mikepavlos committed:
Merge pull request #120 from online-furniture-store/fix/popular-products-image-display

fix: fix/popular-products-image-display исправляет отображение изобра…

Markello93 committed:
Merge branch 'dev' into feat/product

Markello93 committed:
main_image_admin

dependabot[bot] committed:
build(deps): bump drf-spectacular from 0.26.3 to 0.26.4

Bumps [drf-spectacular](https://github.com/tfranzel/drf-spectacular) from 0.26.3 to 0.26.4.
- [Release notes](https://github.com/tfranzel/drf-spectacular/releases)
- [Changelog](https://github.com/tfranzel/drf-spectacular/blob/master/CHANGELOG.rst)
- [Commits](https://github.com/tfranzel/drf-spectacular/compare/0.26.3...0.26.4)

---
updated-dependencies:
- dependency-name: drf-spectacular
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump traefik in /compose/production/traefik

Bumps traefik from 2.10.3 to 2.10.4.

---
updated-dependencies:
- dependency-name: traefik
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump flake8 from 6.0.0 to 6.1.0

Bumps [flake8](https://github.com/pycqa/flake8) from 6.0.0 to 6.1.0.
- [Commits](https://github.com/pycqa/flake8/compare/6.0.0...6.1.0)

---
updated-dependencies:
- dependency-name: flake8
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump django-anymail from 10.0 to 10.1

Bumps [django-anymail](https://github.com/anymail/django-anymail) from 10.0 to 10.1.
- [Release notes](https://github.com/anymail/django-anymail/releases)
- [Changelog](https://github.com/anymail/django-anymail/blob/main/CHANGELOG.rst)
- [Commits](https://github.com/anymail/django-anymail/compare/v10.0...v10.1)

---
updated-dependencies:
- dependency-name: django-anymail
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump uvicorn[standard] from 0.23.1 to 0.23.2

Bumps [uvicorn[standard]](https://github.com/encode/uvicorn) from 0.23.1 to 0.23.2.
- [Release notes](https://github.com/encode/uvicorn/releases)
- [Changelog](https://github.com/encode/uvicorn/blob/master/CHANGELOG.md)
- [Commits](https://github.com/encode/uvicorn/compare/0.23.1...0.23.2)

---
updated-dependencies:
- dependency-name: uvicorn[standard]
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Markello93 committed:
Merge pull request #121 from online-furniture-store/feat/product

Feat/product

dependabot[bot] committed:
build(deps): bump drf-extra-fields from 3.5.0 to 3.7.0

Bumps [drf-extra-fields](https://github.com/Hipo/drf-extra-fields) from 3.5.0 to 3.7.0.
- [Release notes](https://github.com/Hipo/drf-extra-fields/releases)
- [Commits](https://github.com/Hipo/drf-extra-fields/compare/v3.5.0...v3.7.0)

---
updated-dependencies:
- dependency-name: drf-extra-fields
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

ReznikovRoman commented on apps/orders/admin.py:
```diff
@@ -0,0 +1,50 @@
+
+class OrderProductInline(admin.TabularInline):
+    model = OrderProduct
+    raw_id_fields = ['product']
```
Странно, что в каких-то местах ты используешь кортеж для перечисления, а в каких-то - список

Советую одного подхода в проекте придерживаться

ReznikovRoman commented on apps/orders/models.py:
```diff
@@ -0,0 +1,116 @@
+
+
+class DeliveryType(models.Model):
+    """Модель способов доставки"""
```
По правилам [pydocstyle](http://www.pydocstyle.org/en/stable/error_codes.html) докстринг должен заканчиваться точкой (D400)

Советую установить для этого специальный плагин к flake8 - https://pypi.org/project/flake8-docstrings/

ReznikovRoman commented on apps/orders/models.py:
```diff
@@ -0,0 +1,116 @@
+    user = models.ForeignKey(
+        User, verbose_name='Пользователь', on_delete=models.CASCADE, related_name='orders', null=True
+    )
+    products = models.ManyToManyField(Product, through='OrderProduct', verbose_name='Товар')
```
Не забывай указывать `related_name` для всех `ForeignKey`/`ManyToMany`

Проверь во всем проекте

ReznikovRoman commented on apps/orders/models.py:
```diff
@@ -0,0 +1,116 @@
+        self.cost = self.price * self.quantity
+        super().save(*args, **kwargs)
+
+        self.order.total_cost = self.order.order_products.aggregate(Sum('cost'))['cost__sum']
```
Общую стоимость нужно однозначно считать на лету при необходимости, а не хранить в БД

Стоимость `cost` из этой модели тоже стоит вынести, но это уже на ваше усмотрение

ReznikovRoman commented on apps/orders/serializers.py:
```diff
@@ -0,0 +1,133 @@
+    @staticmethod
+    def update_storehouse(products):
+        """Проверяет и обновляет кличество на складе после заказа."""
+
```
После докстринга метода/функции не нужно пропускать строку

Проверь во всем проекте

ReznikovRoman commented on apps/product/admin.py:
```diff
@@ -0,0 +1,167 @@
+    list_editable = ('price', 'fast_delivery')
+    inlines = (DiscountInLine, FavoriteInLine, CartItemInLine)
+    search_fields = ('article', 'name', 'brand')
+    list_filter = ('article', 'name', 'category')
```
Зачем фильтрация по названию, если у тебя есть поиск? Тем более с такой фильтрацией придется отрендерить **весь** список продуктов **без пагинации**

Такая фильтрация может быть оправдана для связанных полей (например, `article`), булевых значений или полей с перечислением, enum

Проверь во всей админке

ReznikovRoman commented on apps/product/admin.py:
```diff
@@ -0,0 +1,167 @@
+class ProductAdmin(ImportExportModelAdmin):
+    list_display = (
+        'pk',
+        'article',
```
Со временем в БД накопится большое количество записей, и страница со списком будет долго открываться

Поэтому советую заранее позаботиться об оптимизации: переопределить метод `get_queryset` и заджойнить/зафетчить связанные поля. Например, здесь можно заджойнить категорию, article, и другие связанные поля ForeignKey, а зафетчить ManyToMany поля

Проверь во всей админке подобное (например, в корзине заджойнить пользователя)

ReznikovRoman commented on apps/product/cart_views.py:
```diff
@@ -0,0 +1,94 @@
+from apps.product.models import CartItem, CartModel, Product
+
+
+@extend_schema(responses={status.HTTP_200_OK: CartModelSerializer}, methods=['GET'])
```
Чтобы не занимать место в файле с контроллерами, описание OpenAPI обычно выносят в отдельный файл. Примерно так - https://github.com/ReznikovRoman/netflix-admin/blob/master/netflix/api/v1/movies/openapi.py

Проверь во всем проекте

ReznikovRoman commented on apps/product/cart_views.py:
```diff
@@ -0,0 +1,94 @@
+    methods=['DELETE'],
+)
+@api_view(['DELETE'])
+def del_item(request, id):
```
`del`: Не нужно сокращать названия переменных/методов подобным образом. Обо всех общепринятых сокращениях я расскажу в ревью, если такие встретятся

Проверь во всем проекте

ReznikovRoman commented on apps/product/models.py:
```diff
@@ -0,0 +1,303 @@
+        ordering = ('purpose',)
+
+    def __str__(self):
+        fields = [
```
nit: Вам точно нужно такое сложное строковое представление для модели? Их просто наоборот стараются сделать как можно проще и короче

Проверь во всем проекте

ReznikovRoman commented on apps/users/models.py:
```diff
@@ -1,35 +1,56 @@
-    Default custom user model for online-furniture-store-backend.
-    If adding fields that need to be filled at user signup,
-    check forms.SignupForm and forms.SocialSignupForms accordingly.
+    Настраиваемая модель пользователя.
```
Докстринг надо стараться умещать на одной строке (без переноса на следующую)

То есть примерно так:

```python
"""Настраиваемая модель пользователя."""
```

Проверь во всем проекте

ReznikovRoman commented on apps/orders/views.py:
```diff
@@ -0,0 +1,73 @@
+class DeliveryViewSet(viewsets.ModelViewSet):
+    """Вьюсет для доставок."""
+
+    queryset = Delivery.objects.all()
```
Нужно еще заджойнить тип доставки `type_delivery`

Проверь во всех контроллерах корректные ОРМ запросы с оптимизацией (select_related для ForeignKey/OneToOne, prefetch_related для ManyToMany/reverse ForeignKey для полей, которые используются в сериалайзерах)

ReznikovRoman reviewed (changes requested):
Хорошо поработали! Сейчас совсем немного замечаний

dependabot[bot] committed:
build(deps): bump python in /compose/production/django

Bumps python from 3.11.4-slim-bullseye to 3.11.5-slim-bullseye.

---
updated-dependencies:
- dependency-name: python
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump python in /compose/local/django

Bumps python from 3.11.4-slim-bullseye to 3.11.5-slim-bullseye.

---
updated-dependencies:
- dependency-name: python
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump docker/build-push-action from 4 to 5

Bumps [docker/build-push-action](https://github.com/docker/build-push-action) from 4 to 5.
- [Release notes](https://github.com/docker/build-push-action/releases)
- [Commits](https://github.com/docker/build-push-action/compare/v4...v5)

---
updated-dependencies:
- dependency-name: docker/build-push-action
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump docker/login-action from 2 to 3

Bumps [docker/login-action](https://github.com/docker/login-action) from 2 to 3.
- [Release notes](https://github.com/docker/login-action/releases)
- [Commits](https://github.com/docker/login-action/compare/v2...v3)

---
updated-dependencies:
- dependency-name: docker/login-action
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

SaPer663 committed:
Merge pull request #134 from online-furniture-store/dependabot/github_actions/docker/login-action-3

build(deps): bump docker/login-action from 2 to 3

dependabot[bot] committed:
build(deps): bump docker/setup-buildx-action from 2 to 3

Bumps [docker/setup-buildx-action](https://github.com/docker/setup-buildx-action) from 2 to 3.
- [Release notes](https://github.com/docker/setup-buildx-action/releases)
- [Commits](https://github.com/docker/setup-buildx-action/compare/v2...v3)

---
updated-dependencies:
- dependency-name: docker/setup-buildx-action
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

SaPer663 committed:
Merge pull request #133 from online-furniture-store/dependabot/github_actions/docker/setup-buildx-action-3

build(deps): bump docker/setup-buildx-action from 2 to 3

SaPer663 committed:
Merge pull request #132 from online-furniture-store/dependabot/github_actions/docker/build-push-action-5

build(deps): bump docker/build-push-action from 4 to 5

dependabot[bot] committed:
build(deps): bump actions/checkout from 3 to 4

Bumps [actions/checkout](https://github.com/actions/checkout) from 3 to 4.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v3...v4)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

SaPer663 committed:
Merge pull request #131 from online-furniture-store/dependabot/github_actions/actions/checkout-4

build(deps): bump actions/checkout from 3 to 4

SaPer663 committed:
Merge pull request #130 from online-furniture-store/dependabot/docker/compose/local/django/python-3.11.5-slim-bullseye

build(deps): bump python from 3.11.4-slim-bullseye to 3.11.5-slim-bullseye in /compose/local/django

SaPer663 committed:
Merge pull request #129 from online-furniture-store/dependabot/docker/compose/production/django/python-3.11.5-slim-bullseye

build(deps): bump python from 3.11.4-slim-bullseye to 3.11.5-slim-bullseye in /compose/production/django

SaPer663 committed:
Merge pull request #128 from online-furniture-store/dependabot/pip/drf-extra-fields-3.7.0

build(deps): bump drf-extra-fields from 3.5.0 to 3.7.0

SaPer663 committed:
Merge branch 'dev' into dependabot/pip/uvicorn-standard--0.23.2

SaPer663 committed:
Merge pull request #127 from online-furniture-store/dependabot/pip/uvicorn-standard--0.23.2

build(deps): bump uvicorn[standard] from 0.23.1 to 0.23.2

SaPer663 committed:
Merge pull request #122 from online-furniture-store/dependabot/pip/drf-spectacular-0.26.4

build(deps): bump drf-spectacular from 0.26.3 to 0.26.4

SaPer663 committed:
Merge pull request #123 from online-furniture-store/dependabot/docker/compose/production/traefik/traefik-2.10.4

build(deps): bump traefik from 2.10.3 to 2.10.4 in /compose/production/traefik

SaPer663 committed:
Merge pull request #124 from online-furniture-store/dependabot/pip/flake8-6.1.0

build(deps): bump flake8 from 6.0.0 to 6.1.0

SaPer663 committed:
Merge pull request #126 from online-furniture-store/dependabot/pip/django-anymail-10.1

build(deps): bump django-anymail from 10.0 to 10.1
